src/XMakeTasks/Microsoft.CSharp.CurrentVersion.targets

```diff
@@ -46,6 +46,18 @@ Copyright (C) Microsoft Corporation. All rights reserved.
         <SynthesizeLinkMetadata Condition=" '$(SynthesizeLinkMetadata)' == '' and '$(HasSharedItems)' == 'true' ">true</SynthesizeLinkMetadata>
     </PropertyGroup>
 
+    <PropertyGroup>
+        <!-- Select a default C# compiler here, if unspecified, instead of in the Csc task, so that we can also
+             decide on the debug extension to be used. -->
+        <CscToolExe Condition="'$(CscToolExe)' == '' and '$(MSBuildRuntimeType)' != 'Mono'">csc.exe</CscToolExe>
+        <CscToolExe Condition="'$(CscToolExe)' == '' and '$(MSBuildRuntimeType)' == 'Mono'">mcs.exe</CscToolExe>
```

Comment:
Why? I don't understand what using the Mono runtime has to do with using `mcs`. The compiler should be independent of the runtime.